forte/fci/fci_solver_initial_guess.cc

```diff
@@ -63,14 +63,26 @@ std::vector<Determinant> FCISolver::initial_guess_generate_dets(std::shared_ptr<
                 vec_e_I.begin(), vec_e_I.end(),
                 [&e](const std::tuple<double, size_t>& t) { return e < std::get<0>(t); });
             vec_e_I.insert(it, std::make_tuple(e, I));
-            emax = std::get<0>(vec_e_I.back());
+            // Do not update the maximum energy threshold if using core determinants as initial guess
+            if (!(core_guess_)){
+                emax = std::get<0>(vec_e_I.back());
+            }
             added++;
         }
     }
 
     std::vector<Determinant> guess_dets;
     for (const auto& [e, I] : vec_e_I) {
-        guess_dets.push_back(lists_->determinant(I, symmetry_));
+        const auto& det = lists_->determinant(I, symmetry_);
+        // If using core determinants as initial guess include those with
+        // single and double holes in first bit position
+        if (core_guess_){
+            if (!(det.get_alfa_bit(0) and det.get_beta_bit(0))){
+                guess_dets.push_back(det);
+            }
+        } else {
+            guess_dets.push_back(det);
```

Comment:
Let us change push_back to emplace_back here, though the speed does not matter at this place. I can barely find a place where emplace_back is worse than push_back